src/services/bunker_cases/midterm_slashing_penalty.py

```diff
@@ -84,104 +77,148 @@ def get_per_epoch_buckets(
         https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/beacon-chain.md#modified-slash_validator
         https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#initiate_validator_exit
         """
-        per_epoch_buckets: dict[EpochNumber, dict[str, Validator]] = defaultdict(dict[str, Validator])
-        for key, validator in all_slashed_validators.items():
-            v = validator.validator
-            if not v.slashed:
-                raise ValueError("Validator should be slashed to detect slashing epoch range")
-            if int(v.withdrawable_epoch) - int(v.exit_epoch) > MIN_VALIDATOR_WITHDRAWABILITY_DELAY:
-                determined_slashed_epoch = EpochNumber(int(v.withdrawable_epoch) - EPOCHS_PER_SLASHINGS_VECTOR)
-                per_epoch_buckets[determined_slashed_epoch][key] = validator
-                continue
+        v = validator.validator
 
-            possible_slashed_epoch = int(v.withdrawable_epoch) - EPOCHS_PER_SLASHINGS_VECTOR
-            for epoch in range(max(0, ref_epoch - EPOCHS_PER_SLASHINGS_VECTOR), possible_slashed_epoch + 1):
-                per_epoch_buckets[EpochNumber(epoch)][key] = validator
+        if int(v.withdrawable_epoch) - int(v.exit_epoch) > MIN_VALIDATOR_WITHDRAWABILITY_DELAY:
+            determined_slashed_epoch = EpochNumber(int(v.withdrawable_epoch) - EPOCHS_PER_SLASHINGS_VECTOR)
+            return [determined_slashed_epoch]
 
-        return per_epoch_buckets
+        earliest_slashed_epoch = max(0, ref_epoch - EPOCHS_PER_SLASHINGS_VECTOR)
```

Comment:
may be `earliest_slashed_epoch` -> `earliest_relevant_slashed_epoch` since it could be earlier, but just doesn't matter for midterm calcs

Reply:
fixed [here](https://github.com/lidofinance/lido-oracle/pull/261/commits/a8f2b0963e05fff55623cb32bc398da9e50c3458)